src/main/scala/pirate/Usage.scala

```diff
@@ -61,38 +88,51 @@ object Render {
       case BothName(s, l) => s"-${s}|--${l}"
     }
 
-    def option(f: Name, metas: List[String]): String =
-      flag(f) + " " + metas.mkString(" ")
+    def flagO(f: Name): String = f match {
+      case ShortName(s) => s"-${s}"
+      case LongName(l) => s"--${l}"
+      case BothName(s, l) => s"(-${s}|--${l})"
+    }
+
+    def option(f: Name, meta: Option[String]): String =
+      flag(f) + meta.cata(x => " " + x, "")
+
+    def mDfault(mark: Boolean)(opt: String): String =
+      if (mark) s"[${opt}]" else opt
+
+    def availableOptions = if (i.switches.length == 0) "" else
+      s"""|Available options:
+          |${flagspace}${(i.switches.map(flaginfo) ++ i.flags.map(optioninfo)).mkString("\n" + flagspace)}
+          |""".stripMargin
 
-    s"""|Usage:
-        |${flagspace}${name} ${synopsis}
+    def availableCommands = if (i.commands.length == 0) "" else
+      s"""|Available commands:
+          |${flagspace}${(i.commands.map(commandinfo)).mkString("\n" + flagspace)}
+          |""".stripMargin
+
+    def full = s"""|Usage:
+        |${flagspace}${wrap(name,mode.flagIndent)(synopsis, mode.width - name.length, name.length + mode.flagIndent + 1)}
         |
         |${description.map(_ + "\n").getOrElse("")}
-        |Options:
-        |${flagspace}${(i.switches.map(flaginfo) ++ i.flags.map(optioninfo)).mkString("\n" + flagspace)}
+        |${availableOptions}
+        |${availableCommands}
         |""".stripMargin
   }
 }
 
 
-case class Info(
-  switches: List[SwitchInfo],
-  flags: List[FlagInfo],
-  arguments: List[ArgumentInfo],
-  commands: List[CommandInfo]
-)
-
-case class SwitchInfo(flag: Name, description: Option[String])
-case class FlagInfo(flag: Name, description: Option[String], metas: List[String], dfault: Boolean)
-case class ArgumentInfo(meta: List[String])
-case class CommandInfo(name: String, description: Option[String])
-object Info {
-  implicit def InfoMonoid: Monoid[Info] = new Monoid[Info] {
-    def zero = Info(Nil, Nil, Nil, Nil)
-    def append(i1: Info, i2: => Info) = Info(i1.switches ++ i2.switches, i1.flags ++ i2.flags, i1.arguments ++ i2.arguments, i1.commands ++ i2.commands)
-  }
+case class Infos(l: List[Info]) {
+  def switches  = l.collect { case a: SwitchInfo   => a }
+  def flags     = l.collect { case a: FlagInfo     => a }
+  def arguments = l.collect { case a: ArgumentInfo => a }
+  def commands  = l.collect { case a: CommandInfo  => a }
 }
 
+sealed trait Info
+case class SwitchInfo(flag: Name, description: Option[String], dfault: Boolean) extends Info
+case class FlagInfo(flag: Name, description: Option[String], meta: Option[String], dfault: Boolean) extends Info
+case class ArgumentInfo(meta: Option[String], description: Option[String], multi: Boolean) extends Info
```

Comment:
Can multi be implied via the tree context? I remember leaving it off intentionally, can't remember the precise reason now though.